### PR TITLE
Fix cache race condition in v2

### PR DESCRIPTION
**Pull request by abhishek58g**
Commit for issue #307

### CODE REVIEW

File: tests/Test.ADAL.Common/TokenCacheTests.cs [hunk 1]
```diff
@@ -665,5 +665,33 @@ public static AuthenticationResult GenerateRandomCacheValue(int maxFieldSize, st
                     UserInfo = new UserInfo { UniqueId = uniqueId, DisplayableId = displayableId }
                 };
         }
+
+        public static void ParallelStorePositiveTest(byte[] oldcache)
+        {
+            for (int i = 0; i < 100; i++)
```
Comment (kpanwar) [inline tests/Test.ADAL.Common/TokenCacheTests.cs]: no need for 100 times.

Comment (abhishek58g) [inline tests/Test.ADAL.Common/TokenCacheTests.cs]: Removed the loop

**Comment by kpanwar** [comment]
you should put lock around all the methods. This will reduce bugs in long run.

**Comment by abhishek58g** [comment]
Added locks for all methods except private ones

**Comment by kpanwar** [comment]
I think you should consider putting a lock around all methods.

**Comment by abhishek58g** [comment]
Added locks around all the methods

**Pull request merged**